volatility3/framework/symbols/windows/extensions/__init__.py

```diff
@@ -492,9 +492,41 @@ def get_name(self) -> str:
         return header.NameInfo.Name.String  # type: ignore
 
 
-class ETHREAD(objects.StructType):
+class ETHREAD(objects.StructType, pool.ExecutiveObject):
     """A class for executive thread objects."""
 
+    def is_valid(self) -> bool:
+        """Determine if the object is valid."""
+
+        try:
+            # validation by TID:
+            if self.Cid.UniqueThread % 4 != 0:  # NT tids are divisible by 4
+                return False
+
+            # validation by PID of parent process:
+            if self.Cid.UniqueProcess % 4 != 0:
+                return False
+
+            # validation by thread creation time:
+            if (
+                self.Cid.UniqueProcess != 4
+            ):  # The System process (PID 4) has no create time
+                ctime = self.get_create_time()
+                if not isinstance(ctime, datetime.datetime):
+                    return False
+
+        except exceptions.InvalidAddressException:
+            return False
+
+        # passed all validations
+        return True
+
+    def get_create_time(self):
```

Comment:
I noticed when testing this plugin that the creation timestamps were off, although the exit times were as expected for this sample:

```
0x1a03a30	1436	1112	0x7c8106e9	4891-04-14 17:32:21.000000 	2012-04-14 23:11:48.000000 
0x1a03da8	696	1244	0x7c8106e9	4891-04-14 21:24:12.000000 	2012-04-14 23:40:31.000000 
0x1a05da8	1096	464	0x7c8106e9	4891-04-14 21:24:12.000000 	2012-04-14 23:40:31.000000 
```

For some reason, these two timestamps are not in the same format. Check out how vol2 defines CreateTime as a "ThreadCreateTimeStamp" (https://github.com/volatilityfoundation/volatility/blob/master/volatility/plugins/overlays/windows/windows.py#L69). 

The underlying difference is that some bits are shifted for the creation timestamp: https://github.com/volatilityfoundation/volatility/blob/master/volatility/plugins/overlays/windows/windows.py#L1033

Reply:
Hi,
Thanks for bringing this serious issue to my attention. I will certainly look into it, and try to address the problem very soon. In fact, I might be able to work on it this weekend.
In the meantime, thx again for your response and feedback! (:

Reply:
Hello @iMHLv2.

After thoroughly looking into the thread creation timestamps and examining the implementation in the vol2 project, I tested on various win7 and win10 mem images (widely-known win7 sample 'stuxnet.vmem', as well as snapshots taken from local Windows virtual machines). 
Unfortunately, I was unable to replicate the issue you encountered, as the timestamps appeared to be accurate. It appears that the existing implementation of the plugin is functioning correctly, however there may still be differences between the memory sample you tested and the ones I tested.

I would be happy if you could please provide the memory sample that has caused the problem, so I could test the plugin on it and further look into the issue.

Thank you for assistance and feedback!

Reply:
Hi @iMHLv2 and @ikelos,
Do you have any update on this? 
It would immensely help me to address your request if you could provide the memory sample that you encountered the bug with, or some guidance or instruction on how to set up my environment to get an equal dump, so I could reproduce the problem you mentioned with the plugin on my end.
Really looking forward to cooperate and contribute to the project!

Reply:
Hi @RuBublik - Sorry for the delay. I couldn't find the exact sample, but I do have several others with the same problem. It appears to be limited to XP samples. Here's an example to play with:

https://volexity-memory-sample-share.s3.amazonaws.com/grrcon.img.zip

Reply:
Hi @iMHLv2 ,
Thank you so much for your response and pinpointing of the problem!
I have tested the plugin against your sample, and indeed reproduced the problem. You were right about the bit shifting, it is present on older windows versions, like XP.
I have revised the code closely replicating the implementation in vol2 you mentioned. After thoroughly testing this `thrdscan` plugin of vol3 in comparison to the counterpart in vol2 on several memory samples of various windows versions - I'm sure the timestamps are correct now.

Looking forward to your review!
Hope that we can proceed with merging this time (-:

Reply:
Hi @iMHLv2,
I've addressed the changes you requested in the pull request. Any chance you could take a moment to review them? 
Looking forward to your approval and merging the PR. 
Thanks!

Reply:
@RuBublik It looks good now, thanks and sorry for the delay.